### ISSUE TITLE
Observability improvements (esp. regarding to context and chat/commands)

### ISSUE THREAD

**Issue by kalanchan** [sourcegraph/cody#3470]
```[tasklist]
### tasks
- [ ] https://github.com/sourcegraph/cody/issues/3687
- [ ] https://github.com/sourcegraph/cody/issues/3688
- [ ] https://github.com/sourcegraph/sourcegraph/issues/60741 (for vis, but not necessarily for client team)
- [ ] https://github.com/sourcegraph/cody/issues/3712
- [ ] https://github.com/sourcegraph/cody-issues/issues/19
```

**Comment by kalanchan** [comment]
@philipp-spiess please work with @jtibshirani and @chillatom to understand where we need context telemetry

**Comment by philipp-spiess** [comment]
We've been doing a bunch of work there in the past few days.

@umpox has been adding a lot of logging to commands, we should now be able to compute how helpful these are and how long edits stay in the document.

I added a small thing where we can now see when users interact with context items in the UI: https://github.com/sourcegraph/cody/pull/3581

One thing remains open and that is to include information on what data made it into the prompt generation on the clients. We started a [thread in Slack](https://sourcegraph.slack.com/archives/C05AGQYD528/p1711022887912369) and it seems desired to add a mechanic similar to what we do on the Autocomplete side where we create a summary event that contains the different retriever sources and how many items were included etc. See: https://github.com/sourcegraph/cody/blob/4201825a4e35fc4d0e3d2b4eb4ae53c207a123c2/vscode/src/completions/context/context-mixer.ts#L21-L47

@kalanchan Unfortunlaty I was not able to finish that in time before leaving for my PTO. I’m happy to add this when I’m back (April 8) but if we need this sooner, it might make sense to find someone else.
